<launch>
<!-- To post-process the obs stat code on a provided list of bags-->
  <arg name="frame"/>
  <arg name="resolution"/>
  <arg name="range_max"/>
  <arg name="bag_list"/>
  <arg name="output_file"/>
  <arg name="xmin" default="-30."/>
  <arg name="xmax" default="30."/>
  <arg name="ymin" default="-30."/>
  <arg name="ymax" default="30."/>
  <arg name="zmin" default="0."/>
  <arg name="zmax" default="6."/>
  <node pkg="obs_stat" name="obs_stat_grid_pub" type="obsstat_grid_pub" output="screen">
    <remap from="~/input_scan" to="/lidar_when_moving/noisy_point_cloud"/>
    <remap from="~/cmd" to="/husky_velocity_controller/cmd_vel"/>
    <param name="map_frame" type="string" value="$(arg frame)"/>
    <param name="is_simu" type="string" value="true"/>
    <param name="only_when_moving" type="bool" value="true"/>
    <param name="range_max" value="$(arg range_max)"/>
    <param name="filter_negative_z" type="bool" value="false"/>
    <param name="is_3d" type="bool" value="true"/>
    <param name="grid_xmin" value="$(arg xmin)"/>
    <param name="grid_xmax" value="$(arg xmax)"/>
    <param name="grid_ymin" value="$(arg ymin)"/>
    <param name="grid_ymax" value="$(arg ymax)"/>
    <param name="grid_zmin" value="$(arg zmin)"/>
    <param name="grid_zmax" value="$(arg zmax)"/>
    <param name="bottom_thres" value=".2"/>
    <param name="resolution_m_per_pix" value="$(arg resolution)"/>
    <param name="bag_list" value = ""/>
    <param name="topic_name" value = "/lidar_when_moving/noisy_point_cloud"/>
    <param name="output_file" value = ""/>
  </node>
</launch>
